### DISCUSSION TITLE
Question about usage of Meter

### DISCUSSION THREAD

**Discussion by vasigorc** [open-telemetry/opentelemetry-java#4227] [Q&A] [answered]
Hello 👋 

We are implementing metrics collection using Opentelemetry Java SDK. I wanted to confirm that my understanding of using the SDK is correct.

First here is a [thread from otel-specification board](https://github.com/open-telemetry/opentelemetry-specification/discussions/2310) which includes our rough class diagram. Basically we use SDK Provider and Otlp Grpc Exporter.

Our app is written in Scala, but I think it is comprehensible:

```scala
case TimerMetric(metricName, labels, duration) =>
      val timerRecord = meter
        .longValueRecorderBuilder(metricName.value)
        .setUnit("Milliseconds")
        .build()

      timerRecord.record(duration.toMillis, labelifyAttributes(labels))
``` 

☝️  `TimerMetric` is our own abstraction for collecting metrics that tracks duration and `meter` is an instance of `io.opentelemetry.api.metrics.Meter`. The question is just that - is calling `.record(...)` on the last line sufficient to send the data over the wire with all else being correctly configured?

**Comment by jkwatson** [comment, marked as answer]
This looks fine (although an older version of the API... ValueRecorder has been replaced by Histogram in the stable version of the API).

But, a lot of it depends on how you've configured the SDK that backs the API calls.

**Reply by vasigorc** [reply]
Thanks John! Yes, I've upgraded to `1.10` so the code more looks like this:

```scala
  override def wrappedReceive: Receive = {
    case TimerMetric(metricName, labels, duration) =>
      meter
      .gaugeBuilder(metricName.value)
      .setUnit(unit)
      .buildWithCallback(measurement => {
        measurement.record(duration.toMillis.toDouble, toOtelAttributes(labels))
      })
```
 > ValueRecorder has been replaced by Histogram in the stable version of the API)
 
 I was leaning towards a Gauge since Histogram is said to be an aggregation. In our case it is a snapshot.
 
 Also from [Dropwizard](https://metrics.dropwizard.io/4.2.0/manual/core.html): "A Histogram measures the distribution of values in a stream of data"

**Reply by jkwatson** [reply]
You don't want to get a distribution of timings? The Histogram should be perfect for that.

**Reply by vasigorc** [reply]
That would actually be quite helpful, thanks for mentioning that! I'll mark this as answered 👍
